methylpy/utilities.py

```diff
@@ -681,34 +682,46 @@ def index_allc_file(allc_file,reindex=False):
             pass
         else:
             return 0
-    g = open(index_file,'w')
-    f = open_allc_file(allc_file)
-    cur_chrom = ""
+
+    if allc_file.endswith('gz'):  # works for .gz, .bgz
+        f = subprocess.Popen(['zcat', allc_file],
+                             stdout=subprocess.PIPE,
+                             encoding='utf8').stdout
+    else:
+        f = open(allc_file)
```

Comment:
A probably better way to make these changes is to modify the `open_allc_file` function while keeping `f = open_allc_file(allc_file)`

https://github.com/yupenghe/methylpy/blob/4f634eeb26d718f14f4251c59e18b9d0a1bf9517/methylpy/utilities.py#L1134-L1135

Reply:
like this? I use standalone Popen because its faster then gzip.open (due to create a subprocess). In my own project, I write a more [sophesticated gzip openner ](https://github.com/lhqing/cemba_data/blob/master/cemba_data/tools/open.py) learnt from [cutadapt's gzip openner](https://github.com/marcelm/xopen)

```python

def open_allc_file(allc_file):
    if allc_file.endswith('gz'):  # so it works for .gz and .bgz
        f = subprocess.Popen(['zcat', allc_file],
                             stdout=subprocess.PIPE,
                             encoding='utf8').stdout
    else:
        f = open(allc_file,'r')
    return(f)

```

Reply:
I use `open_allc_file` in `index_allc_file` and modified the `open_allc_file`
Can you check again?